eos-live-boot-overlayfs-setup

```diff
@@ -88,6 +88,64 @@ setup_and_mount_storage() {
 	mount "${device}" /run/eos-live
 }
 
+setup_ro_mount() {
+    local path=$1
+
+    # If the directory isn't a mount point, make a read only bind mount
+    # on top of the directory and be done.
+    if ! findmnt "$path" >/dev/null; then
+        mount -o bind,ro "$path" "$path"
+        return 0
+    fi
+
+    # In case of an automount, trigger the real mount so its options can
+    # be inspected.
+    ls "$path" >/dev/null
```

Comment:
Is there a timeout on the automount for `/boot`?

Reply:
Systemd uses 120 seconds (2 minutes). I think that should be more than enough time, but if that's not deemed sufficient, a simple way to keep the mount from going idle is to change to the directory. I.e., `pushd $path; ...; popd` would do it.

An alternative I considered was changing the timeout to 0 so that the automounts never consider idleness. Unfortunately, you'd either have to restart the automount systemd unit after changing its parameters or issue an `ioctl` to the existing autofs mount. In either case you'd have to unmount a potentially mounted real mount first. autofs in all its wisdom uses an open fd on the root directory _of the autofs mount, not the real mount_ to issue `ioctl`s. Unless you've stashed an open fd from before the real mount shadows the autofs mount, there's no way to make an `ioctl` on an autofs mount.

Reply:
I think it's fine as-is – realistically the script will not take 2 minutes.

Reply:
It also just needs to survive the subsequent `mounted_options=$(findmnt -n -f -d backward -o OPTIONS "$path")` line. The rest of the function behaves correctly whether the mount has been triggered or not.